src/DynamoCore/Core/Threading/DynamoScheduler.cs

```diff
@@ -74,6 +74,18 @@ public interface IScheduler
         IEnumerable<AsyncTask> Tasks { get; }
     }
 
+    public enum TaskProcessMode
+    {
+        /// <summary>
+        /// Scheduled task will be processed immediately
+        /// </summary>
+        Immediately,
+        /// <summary>
+        /// Scheduled task will be processed by schedule
+        /// </summary>
+        BySchedule
+    }
+
```

Comment:
`Is` sounds redundant here, maybe it's just me. I'd rather it used as `TaskProcessMode.Asynchronous` than `TaskProcessMode.IsAsynchronous`. But, your call :)